Neos.Cache/Classes/Backend/SimpleFileBackend.php

```diff
@@ -520,7 +585,9 @@ protected function readCacheFile(string $cacheEntryPathAndFilename, int $offset
      */
     protected function writeCacheFile(string $cacheEntryPathAndFilename, string $data)
     {
-        for ($i = 0; $i < 3; $i++) {
+        $microtime = microtime(true);
+        $acceptedLatencyInSeconds = $this->maxAcceptedWriteLatencyMicroSeconds / 1000000;
```

Comment:
We should probably also calculate the max random sleep based on the max latency. As is, this currently does `10.000 / ((500+10) / 2) = 40` attempts on average and 1.000 in the worst case with default config, which feels a bit much. Also, this increases if the max wait is increased and effectively ends in a busy wait cycle. If we limit max to e.g. `maxAcceptedWriteLatencyMicroSeconds / 5` we have on avg. 10 attempts (but still 1.000 in the unlikely worst case). Or we also scale the minimum usleep scaled by `maxAcceptedWriteLatencyMicroSeconds / 20`? Then we have avg. 8 attempts and 20 in the worst case.

Reply:
Uh! Completely right!Valid point. What about adjusting both boundaries depending on the maximal values by calculating an expected value to match a given amount of tries?

something like:
```
        $targetedTries = 5;
        $waitTimeScatter = 0.5;
        $expectedWaitTimeAverage = $acceptedLatencyInSeconds / $targetedTries;
        $halfExpectedWaitTimeAverage = $expectedWaitTimeAverage * $waitTimeScatter;
        $minimalWaitTime = $expectedWaitTimeAverage - $halfExpectedWaitTimeAverage;
        $maximalWaitTime = $expectedWaitTimeAverage + $halfExpectedWaitTimeAverage;
```

And as these calculations will be done often, maybe they should go to the setters, to not repeat them.
And of course, we have to decide if tries and scatter is something which should be configurable

Reply:
Not sure that's straightforward in this probablistic case. Configuring a "targetTries" is probably something that people don't know what to do with or want to know how to configure the allowed deviation (scatter). But let's not overcomplicate it too much (for the user) - being able to configure a max latency is clear, gives perf. guarantees and works. The number of retries can just be picked with a senseful decision and be made explicit.